### ISSUE TITLE
False positive not-an-iterable for first argument staticmethod in metaclass

### ISSUE THREAD

**Issue by ifinik** [PyCQA/pylint#3032]
### Steps to reproduce
Save the following into a file and run `pylint` over it:
```
class My(type):
    @staticmethod
    def test(args):
        """
        Non-iterable value args is used in an iterating contextpylint(not-an-iterable)
        """
        #         \|/
        for i in args:
            print(i)
```

### Current behavior
A `not-an-iterable` error is being emitted for correct code:
`a.py:4:17: E1133: Non-iterable value args is used in an iterating context (not-an-iterable)`

### Expected behavior
No error should be emitted

### pylint --version output
pylint 2.3.1
astroid 2.2.5

**Comment by PCManticore** [comment]
Hey @ifinik Thanks for creating an issue. Please respect the template for new issues which indicates that it should have a reproduction example, the error you are seeing, the version of `pylint` you are using and so on. This time I went ahead and changed the issue for you.

**Comment by ifinik** [comment]
@PCManticore , I apologize, thanks for the correction.
Is there somewhere an example of a template that can be referenced in such a case?
